torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -222,7 +221,12 @@ Tensor pow_backward_exponent(Tensor grad, const Tensor& self, const Tensor& expo
 }
 
 Tensor pow_backward_exponent(Tensor grad, const Scalar & base, const Tensor& exponent, Tensor result) {
-  auto base_ = base.isComplex() ? base.toComplexDouble() : base.toDouble();
+  Scalar base_;
+  if (base.isComplex()) {
+    base_ = base.toComplexDouble();
+  } else {
+    base_ = base.toDouble();
+  }
```

Comment:
This looks really fishy. `base_` is going to be the same Scalar as `base`; what the code is doing here is unwrap the Scalar and then rewrap it into a Scalar

Reply:
yeah updated